test/shared/speechRecognitionTests.jsx

```diff
@@ -7,7 +7,8 @@ const SpeechRecognitionTests = ({
   Example,
   TestComponent,
   mockOnResult,
-  mockOnEnd
+  mockOnEnd,
+  mockOnError
```

Comment:
I see you pass this mockOnError in, but we're not asserting that it gets called specifically.

I would add a test case specifically for it. The tests are a bit messy at the moment because it's supporting both the hooks and render props components from version 1 (I'll probably remove this in my next release because it can be a bit confusing)

This would involve adding some code to the MockRecognition class that wraps the start call in a try catch. The catch then calls onerror with a generated error event. You could then set up a seperate test case that mocks the underlying start function like so

```js
describe('when the user blocks permission', () => {
        beforeAll(() => {
          MockRecognition.start = jest.fn(() => { throw new Error('not allowed'); });
        });

        it.only('calls the onError function', () => {
          expect(mockOnError.mock.calls.length).toBe(1);
        });
      });
```

I'd be happy to add this in myself if it's not too clear how.

Reply:
I'm very new to jest tests. I tried to do as you recommended but the tests fail. Any pointers would be welcome!

Reply:
Looks like a solid attempt considering you’re so new to this. I’ll take a better look today

Reply:
Ok I see what the issue is. The block which calls and tests the listen functionality is here:

https://github.com/MikeyParton/react-speech-kit/blob/f2ace8031d69dcdcddb5e2721dbeda104338bab1/test/shared/speechRecognitionTests.jsx#L56-L65

If you move your test down into this block listen will be called and the error thrown. After this you'll still have an error in the tests for the SpeechRecognition component because they share the same tests. For now you can just add your functionality to that component too. I'll do some cleanup in a seperate PR because it doesn't make sense to keep maintaining 2 versions of these components.

Reply:
Got it. All tests pass now except the one for the SpeechRecognition component, as you said.